README.md
# Portfolio v2 (PHP version)

[v2.pierreprezelin.com](https://v2.pierreprezelin.com) · [Astro version](https://github.com/OldsunFlush/portfolio-v2-astro) 🚀

For easier deployments, codebase has been moved to [Astro](https://astro.build/).

changelog.txt
Changelog :

[v2.8] (08-04-20)
- End of beta, official release
- Adds the complete "Options" menu
- Replaces fancybox.js by PhotoSwipe
- Replaces SoundCloud by CodePen
- Replaces YouTube by GitHub
- Optimizes responsive for large screens
- Hides Steam Early Access section by default
- Updates content
- Updates assets

/* ------------------------------------------
   Previous updates
------------------------------------------ */

[v2.7.1b] (21-11-18)
- Fixes some links

[v2.7b] (21-11-18)
- End of alpha, beta version release
- Transfers assets to a new hosting service
- Makes some graphic elements a bit more modern
- Fixes a z-index bug on map
- Updates assets (and removes obsolete ones)
- Updates the scroller animation for a more modern one
- Removes the PHP LESS compiler for Prepros
- Removes .htaccess configuration

v2.6.3a (13-05-17)
- Loads fonts progressively rather than making them pop in
- Updates content
- Updates assets
- Disables fancybox.js

v2.6.2a (26-10-16)
- Disables the preloader
- Disables tappable phone numbers on mobile
- Adds hover animations on all contact options

v2.6.1a (19-07-16)
- Replaces all translateX-Y for translate3d animations
- Changes contact email adress

[v2.6a] (18-07-16)
- Adds a custom page for the ChangeLog
- Adds a custom page for the Legal Notice
- Updates jQuery to the 3.x version
- Hides the scroll arrows on mobile
- Fixes IE layout bugs

v2.5.5a (17-07-16)
- Includes the prefixer.less library for better browser compatibility
- Converts 'em' and 'px' units into 'rem'
- Adds progressive JPEGs for the "Home" and "Contact" sections
- Fixes an hover state bug in the mobile menu
- Fixes a print bug in the 404 page

v2.5.4a (14-07-16)
- Makes English the default language
- Disables the resume link
- Small responsive corrections and text updates

v2.5.3a (23-03-16)
- Changes mail animation
- Changes site title size
- Changes subtitles size and styles
- Makes the header background darker on scroll
- Deletes the "Full gallery" button

v2.5.2a (21-03-16)
- Reintroduces a preloader overlay with PaceJS
- Adds Google Analytics codes
- Makes header reduction smoother
- Removes the "Please enable your JavaScript" message in preloader

v2.5.1a (01-03-16)
- Redefines update history and roadmap
- Changes the PHP files structure for better maintenance and lisibility
- Corrects meta texts
- Disables the Apple web app capability

[v2.5a] (26-02-16)
- Enables a better file cache and compression for faster loading
- Reinforces website security
- Adds date informations in the change.log file
- Adds blurred background images on large displays (> 1920px) and tablets (< 1024px)
- Adds filters in the "Works" section
- Changes the "Works" section overlay styles
- Changes works list for a fluid grid
- Disables the CSS preloader
- Disables WOWJS transitions
- Disables the cookie notice on smartphones

v2.4.1a (18-02-16 - unknown before)
- Adds a third line of works
- Adds a custom color to the browser top bar (Android)
- Activates the Behance link
- Updates jQuery to 2.2.0 version

[v2.4a]
- Adds the change.log file link in the footer
- Adds the "Settings" menu initial structure and transitions
- Adds a ripple effect on "Skills" section
- Makes the "Back to top" button appears on scroll and disappears on mobile
- Fixes a bug where fancybox.js makes the body offset
- Corrects responsive in footer and 404 page
- Updates contents and animations

v2.3.1a
- Fixes a logo bug in Safari 5.1.7
- Fixes a bug with the "Back to top" button moving up and down
- Updates mixins et prefixes

[v2.3a]
- Adds a parallaxed background to the "Home" section
- Removes backgrounds lines for a CSS pattern
- Makes the cookie notice less intrusive
- Fixes an overlay bug in the "Works" section.
- Changes "send a mail" hover transitions
- Changes the position of the "back to top" button
- Removes some CSS transitions
- Adds custom styles to the 404 page
- Responsive corrections
- General code optimization

v2.2.1a
- Adds the cookie notification
- Adds a "Scroll to top" fixed button
- Texts and works modifications
- General code optimization

[v2.2a]
- Adds PHP multilingual support (FR, EN)

v2.1.3a
- Adds an "Error 404" message
- PNG images optimization
- Text corrections

v2.1.2a
- Imports the resume v2

v2.1.1a
- Changes the "Works" section overlay styles
- General code optimization

[v2.1a]
- Libraries are now called from a CDN, with removal of locale dependencies (Font Awesome, jQuery, LESS)
- Replaces the CSS responsive header overlay by jQuery, with removal of CSS "display" properties
- Replaces the "Skills" and "Early Access" sections slide by jQuery, with removal of CSS "max-height" animations
- Replaces the "Skills" icons by sprites
- Custom fancybox.js styles
- Prepares the removal of fancybox.js for workLoad.js
- Changes the "Contact" section background image
- Replaces the PNG map marker by a SVG one
- Responsive corrections
- Scripts improvements
- General code optimization
- Deletes some images, files and folders

v2.0.1a
- Adds an automatic less compiler (lessphp 0.4.0 -> lessc.inc.php)
- Adds a "Please enable your JavaScript" message in preloader

[v2.0a]
- Second version alpha release (IE10+ compatible, responsive, Retina-ready, gzip and cache enabled)

/* ------------------------------------------
   Older versions
------------------------------------------ */

v1.7.2 (and less)
-> Check the 1st version changelog file

changelog.php
<?php require_once('common.php'); ?>
<?php require_once('includes/header-navless.inc.php'); ?>

        <div class="changelog">
            <h1>ChangeLog<span><?php echo $lang['LANGUAGE_NOTICE_ENGLISH']; ?></span></span></h1>
            <h2 class="current-version">Current version :</h2>
            <div class="version">
                <h3>[2.8]</h3>
                <h4>
                    <em>08-04-20</em>
                </h4>
                <p>
                    <strong>End of beta, official release</strong>
                </p>
                <ul>
                    <li>Adds the complete "Options" menu</li>
                    <li>Replaces fancybox.js by PhotoSwipe</li>
                    <li>Replaces SoundCloud by CodePen</li>
                    <li>Replaces YouTube by GitHub</li>
                    <li>Optimizes responsive for large screens</li>
                    <li>Hides Steam Early Access section by default</li>
                    <li>Updates content</li>
                    <li>Updates assets</li>
                </ul>
            </div>
            
            <h2>Previous updates :</h2>
            <div class="version">
                <h3>[2.7.1b]</h3>
                <h4>
                    <em>21-11-18</em>
                </h4>
                <ul>
                    <li>Fixes some links</li>
                </ul>
            </div>
            <div class="version">
                <h3>[2.7b]</h3>
                <h4>
                    <em>21-11-18</em>
                </h4>
                <p>
                    <strong>End of alpha, beta version release</strong>
                </p>
                <ul>
                    <li>Transfers assets to a new hosting service</li>
                    <li>Makes some graphic elements a bit more modern</li>
                    <li>Fixes a z-index bug on map</li>
                    <li>Updates assets (and removes obsolete ones)</li>
                    <li>Updates the scroller animation for a more modern one</li>
                    <li>Removes the PHP LESS compiler for Prepros</li>
                    <li>Removes .htaccess configuration</li>
                </ul>
            </div>
            <div class="version">
                <h3>2.6.3a</h3>
                <h4><em>13-05-17</em></h4>
                <ul>
                    <li>Loads fonts progressively rather than making them pop in</li>
                    <li>Updates content</li>
                    <li>Updates assets</li>
                    <li>Disables fancybox.js</li>
                </ul>
            </div>
            <div class="version">
                <h3>2.6.2a</h3>
                <h4><em>26-10-16</em></h4>
                <ul>
                    <li>Disables the preloader</li>
                    <li>Disables tappable phone numbers on mobile</li>
                    <li>Adds hover animations on all contact options</li>
                </ul>
            </div>
            <div class="version">
                <h3>2.6.1a</h3>
                <h4><em>19-07-16</em></h4>
                <ul>
                    <li>Replaces all translateX-Y for translate3d animations</li>
                    <li>Changes contact email adress</li>
                </ul>
            </div>
            <div class="version">
                <h3>[2.6a]</h3>
                <h4><em>18-07-16</em></h4>
                <ul>
                    <li>Adds a custom page for the ChangeLog</li>
                    <li>Adds a custom page for the Legal Notice</li>
                    <li>Updates jQuery to the 3.x version</li>
                    <li>Hides the scroll arrows on mobile</li>
                    <li>Fixes IE layout bugs</li>
                </ul>
            </div>
            <div class="version">
                <h3>2.5.5a</h3>
                <h4><em>17-07-16</em></h4>
                <ul>
                    <li>Includes the prefixer.less library for better browser compatibility</li>
                    <li>Converts 'em' and 'px' units into 'rem'</li>
                    <li>Adds progressive JPEGs for the "Home" and "Contact" sections</li>
                    <li>Fixes an hover state bug in the mobile menu</li>
                    <li>Fixes a print bug in the 404 page</li>
                </ul>
            </div>
            <div class="version">
                <h3>2.5.4a</h3>
                <h4><em>14-07-16</em></h4>
                <ul>
                    <li>Makes English the default language</li>
                    <li>Disables the resume link</li>
                    <li>Small responsive corrections and text updates</li>
                </ul>
            </div>
            <div class="version">
                <h3>2.5.3a</h3>
                <h4><em>23-03-16</em></h4>
                <ul>
                    <li>Changes mail animation</li>
                    <li>Changes site title size</li>
                    <li>Changes subtitles size and styles</li>
                    <li>Makes the header background darker on scroll</li>
                    <li>Deletes the "Full gallery" button</li>
                </ul>
            </div>
            <div class="version">
                <h3>2.5.2a</h3>
                <h4><em>21-03-16</em></h4>
                <ul>
                    <li>Reintroduces a preloader overlay with PaceJS</li>
                    <li>Adds Google Analytics codes</li>
                    <li>Makes header reduction smoother</li>
                    <li>Removes the "Please enable your JavaScript" message in preloader</li>
                </ul>
            </div>
            <div class="version">
                <h3>2.5.1a</h3>
                <h4><em>01-03-16</em></h4>
                <ul>
                    <li>Redefines update history and roadmap</li>
                    <li>Changes the PHP files structure for better maintenance and lisibility</li>
                    <li>Corrects meta texts</li>
                    <li>Disables the Apple web app capability</li>
                </ul>
            </div>
            <div class="version">
                <h3>[2.5a]</h3>
                <h4><em>26-02-16</em></h4>
                <ul>
                    <li>Enables a better file cache and compression for faster loading</li>
                    <li>Reinforces website security</li>
                    <li>Adds date informations in the change.log file</li>
                    <li>Adds blurred background images on large displays (> 1920px) and tablets (< 1024px)</li>
                    <li>Adds filters in the "Works" section</li>
                    <li>Changes the "Works" section overlay styles</li>
                    <li>Changes works list for a fluid grid</li>
                    <li>Disables the CSS preloader</li>
                    <li>Disables WOWJS transitions</li>
                    <li>Disables the cookie notice on smartphones</li>
                </ul>
            </div>
            <div class="version">
                <h3>2.4.1a</h3>
                <h4><em>18-02-16 (unknown before)</em></h4>
                <ul>
                    <li>Adds a third line of works</li>
                    <li>Adds a custom color to the browser top bar (Android)</li>
                    <li>Activates the Behance link</li>
                    <li>Updates jQuery to 2.2.0 version</li>
                </ul>
            </div>
            <div class="version">
                <h3>[2.4a]</h3>
                <ul>
                    <li>Adds the change.log file link in the footer</li>
                    <li>Adds the "Settings" menu initial structure and transitions</li>
                    <li>Adds a ripple effect on "Skills" section</li>
                    <li>Makes the "Back to top" button appears on scroll and disappears on mobile</li>
                    <li>Fixes a bug where fancybox.js makes the body offset</li>
                    <li>Corrects responsive in footer and 404 page</li>
                    <li>Updates contents and animations</li>
                </ul>
            </div>
            <div class="version">
                <h3>2.3.1a</h3>
                <ul>
                    <li>Fixes a logo bug in Safari 5.1.7</li>
                    <li>Fixes a bug with the "Back to top" button moving up and down</li>
                    <li>Updates mixins et prefixes</li>
                </ul>
            </div>
            <div class="version">
                <h3>[2.3a]</h3>
                <ul>
                    <li>Adds a parallaxed background to the "Home" section</li>
                    <li>Removes backgrounds lines for a CSS pattern</li>
                    <li>Makes the cookie notice less intrusive</li>
                    <li>Fixes an overlay bug in the "Works" section.</li>
                    <li>Changes "send a mail" hover transitions</li>
                    <li>Changes the position of the "back to top" button</li>
                    <li>Removes some CSS transitions</li>
                    <li>Adds custom styles to the 404 page</li>
                    <li>Responsive corrections</li>
                    <li>General code optimization</li>
                </ul>
            </div>
            <div class="version">
                <h3>2.2.1a</h3>
                <ul>
                    <li>Adds the cookie notification</li>
                    <li>Adds a "Scroll to top" fixed button</li>
                    <li>Texts and works modifications</li>
                    <li>General code optimization</li>
                </ul>
            </div>
            <div class="version">
                <h3>[2.2a]</h3>
                <ul>
                    <li>Adds PHP multilingual support (FR, EN)</li>
                </ul>
            </div>
            <div class="version">
                <h3>2.1.3a</h3>
                <ul>
                    <li>Adds an "Error 404" message</li>
                    <li>PNG images optimization</li>
                    <li>Text corrections</li>
                </ul>
            </div>
            <div class="version">
                <h3>2.1.2a</h3>
                <ul>
                    <li>Imports the resume v2</li>
                </ul>
            </div>
            <div class="version">
                <h3>2.1.1a</h3>
                <ul>
                    <li>Changes the "Works" section overlay styles</li>
                    <li>General code optimization</li>
                </ul>
            </div>
            <div class="version">
                <h3>[2.1a]</h3>
                <ul>
                    <li>Libraries are now called from a CDN, with removal of locale dependencies (Font Awesome, jQuery, LESS)</li>
                    <li>Replaces the CSS responsive header overlay by jQuery, with removal of CSS "display" properties</li>
                    <li>Replaces the "Skills" and "Early Access" sections slide by jQuery, with removal of CSS "max-height" animations</li>
                    <li>Replaces the "Skills" icons by sprites</li>
                    <li>Custom fancybox.js styles</li>
                    <li>Prepares the removal of fancybox.js for workLoad.js</li>
                    <li>Changes the "Contact" section background image</li>
                    <li>Replaces the PNG map marker by a SVG one</li>
                    <li>Responsive corrections</li>
                    <li>Scripts improvements</li>
                    <li>General code optimization</li>
                    <li>Deletes some images, files and folders</li>
                </ul>
            </div>
            <div class="version">
                <h3>2.0.1a</h3>
                <ul>
                    <li>Adds an automatic less compiler (lessphp 0.4.0 -> lessc.inc.php)</li>
                    <li>Adds a "Please enable your JavaScript" message in preloader</li>
                </ul>
            </div>
            <div class="version">
                <h3>[2.0a]</h3>
                <ul>
                    <li>Second version alpha release (IE10+ compatible, responsive, Retina-ready, gzip and cache enabled)</li>
                </ul>
            </div>
            
            <h2>Older versions :</h2>
            <div class="version">
                <h3>[1.7.2-]</h3>
                <ul>
                    <li>Check the 1st version changelog file</li>
                </ul>
            </div>
            
            <div class="back-container">
                <a href="http://pierre-prezelin.com" title="Pierre Prézelin - www.pierre-prezelin.com" class="btn btn-primary"><?php echo $lang['404_LINK']; ?></a>
            </div>
        </div>

    </body>
</html>